inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -60,71 +90,242 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const DeviceMap<Infer
-        unsigned int optimalNum = 0;
+void MultiDeviceExecutableNetwork::WaitFirstNetworkReady() {
+    if (IsActualNetworkReady()) {
+        return;
+    }
+    if (_cpuFuture.valid() && _acceleratorFuture.valid()) {
         try {
-            optimalNum = network->GetMetric(METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)).as<unsigned int>();
-        } catch (const InferenceEngine::Exception &iie) {
-            IE_THROW()
-                    << "Every device used with the Multi-Device should "
-                    << "support OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric. "
-                    << "Failed to query the metric for the " << device << " with error:" << iie.what();
-        }
-        const auto numRequests = (_devicePriorities.end() == itNumRequests ||
-            itNumRequests->numRequestsPerDevices == -1) ? optimalNum : itNumRequests->numRequestsPerDevices;
-        auto& workerRequests = _workerRequests[device];
-        auto& idleWorkerRequests = _idleWorkerRequests[device];
-        workerRequests.resize(numRequests);
-        _inferPipelineTasksDeviceSpecific[device] = std::unique_ptr<ThreadSafeQueue<Task>>(new ThreadSafeQueue<Task>);
-        auto* idleWorkerRequestsPtr = &(idleWorkerRequests);
-        idleWorkerRequests.set_capacity(numRequests);
-        for (auto&& workerRequest : workerRequests) {
-            workerRequest._inferRequest = { network, network->CreateInferRequest() };
-            auto* workerRequestPtr = &workerRequest;
-            IE_ASSERT(idleWorkerRequests.try_push(workerRequestPtr) == true);
-            workerRequest._inferRequest->SetCallback(
-                [workerRequestPtr, this, device, idleWorkerRequestsPtr] (std::exception_ptr exceptionPtr) mutable {
-                    IdleGuard idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
-                    workerRequestPtr->_exceptionPtr = exceptionPtr;
-                    {
-                        auto capturedTask = std::move(workerRequestPtr->_task);
-                        capturedTask();
-                    }
-                    // try to return the request to the idle list (fails if the overall object destruction has began)
-                    if (idleGuard.Release()->try_push(workerRequestPtr)) {
-                        // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                        // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                        Task t;
-                        if (_inferPipelineTasks.try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t));
-                        else if (_inferPipelineTasksDeviceSpecific[device]->try_pop(t))
-                            ScheduleToWorkerInferRequest(std::move(t), device);
-                    }
-                });
+            _networkFirstReady = _cpuFuture.get();
+            SetPerfCounts(_networkFirstReady);
```

Comment:
again, why the network can be originally loaded with the appropriate PERF_COUNT and we are taking the value (YES/NO) from the loaded netwrok

Reply:
AUTO will change executor device in runtime. one device may support PERF_COUNT and the other may not. Will update this PR to get PERF_COUNT of current executor device. 

 We can have follow up PR for better UX/DX of PERF_COUNT for device switching.

Reply:
I load perfcount configure to device in auto model and get configure to check if need get PerformanceCounts in MultiDeviceAsyncInferRequest